overlord/assertstate/assertstate.go

```diff
@@ -55,8 +55,18 @@ func findError(format string, ref *asserts.Ref, err error) error {
 	}
 }
 
+// AutoRefreshSnapDeclarations refetches all the current snap declarations and their prerequisites while also setting
+// the store.RefreshOptions's IsAutoRefresh to reflect that it was automatic.
+func AutoRefreshSnapDeclarations(s *state.State, userID int) error {
```

Comment:
I will leave this to @pedronis  to decide so maybe best wait for his opinion on this: I personally would slightly prefer to have an options struct on RefreshSnapDecleartions() instead of having two functions. Something like (names are strawmans right now):
```go
type RefreshSnapDeclarationsOptions struct {
    IsAutoRefresh bool
}
func RefreshSnapDeclarations(s *state.State, userID int, opts *RefreshSnapDeclarationsOptions) error {
    if opts == nil { opts = &RefreshSnapDeclarationsOptions{} };
...
```
and then pass this opts down into refreshSnapDeclarations, bulkRefreshSnapDeclarations etc.

But again, I will leave it to Samuele to make the final call on this. I think we want some kind of struct even if we keep two {Auto,}RefreshSnapDeclartions because bools in function declarations are a bit hard to read often. I.e. something like "SetAutomatic(true|false)" is easy to understand but for "refreshSnapDecarations(s, userID, false)" it's not obvious from reading the code in isolation what the "false" stands for and one needs to go back to the declaration of the method etc.

Reply:
I think what was chosen, single function with options is fine. Pawel can tweak this further it later if needed